// Copyright 2022 The Feathub Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package protobuf;

message FeatureValue {
    oneof FeatureValueOneOf {
        bytes bytes_value = 1;
        string string_value = 2;
        int32 int_value = 3;
        int64 long_value = 4;
        double double_value = 5;
        float float_value = 6;
        bool boolean_value = 7;
        BytesArray bytes_array_value = 8;
    }
}

message BytesArray {
    repeated bytes value = 1;
}
